fields/types/gcsimage/GcsImageField.js

```diff
@@ -1 +1,203 @@
+import Field from '../Field';
+import React from 'react';
+import ReactDOM from 'react-dom';
+import classnames from 'classnames';
+import get from 'lodash/get';
+import { Button, FormField, FormInput, FormNote } from 'elemental';
+
+const _ = {
+  get,
+};
+
+const SUPPORTED_TYPES = ['image/gif', 'image/png', 'image/jpeg', 'image/bmp', 'image/x-icon', 'application/pdf', 'image/x-tiff', 'image/x-tiff', 'application/postscript', 'image/vnd.adobe.photoshop', 'image/svg+xml'];
+
+module.exports = Field.create({
+
+  displayName: 'GcsImageField',
+
+  getThumbnail () {
+    return _.get(this.props, 'value.resizedTargets.mobile.url', '');
+  },
+
+  getLocation () {
+    const gcsBucket = _.get(this.props, 'value.gcsBucket', '');
+    const gcsDir = _.get(this.props, 'value.gcsDir', '');
+    const filename = _.get(this.props, 'value.filename', '');
+
+    if (gcsBucket && gcsDir && filename) {
+      if (typeof gcsDir === 'string' && gcsDir.slice(-1) === '/') {
+        return `gs://${gcsBucket}/${gcsDir}${filename}`;
+      }
+      return `gs://${gcsBucket}/${gcsDir}/${filename}`;
+    } else if (gcsBucket && filename) {
+      return `gs://${gcsBucket}/${filename}`;
+    }
+
+    return '';
+  },
+
+  /**
+   *  is this image uploaded from end user local workspace in this session?
+   */
+  hasLocal () {
+    return this.state.origin === 'local';
+  },
+
+  hasImage () {
+    return this.getLocation() || this.hasLocal();
+  },
+
+  getImageSource () {
+    if (this.hasLocal()) {
+      return this.state.localSource;
+    } else if (this.getThumbnail()) {
+      return this.getThumbnail();
+    } else {
+      return null;
+    }
+  },
+
+  /**
+   * Render an image preview
+   */
+  renderImagePreview () {
+    const thumbnail = this.getImageSource();
+
+    if (thumbnail) {
+      return (
+        <div className="image-preview">
+          <a className="img-thumbnail" href={thumbnail} target="_blank">
+            <img className="img-load" style={{ height: '90' }} src={thumbnail} alt="圖片無法載入，請稍後再試。"/>
+          </a>
+        </div>
+      );
+    }
+
+    return null;
+  },
+
+  /**
+   * Render image details
+   */
+  renderImageDetails () {
+    if (this.getLocation()) {
+      const iptc = _.get(this.props, 'value.iptc', {});
+      let createdTime = _.get(iptc, 'created_time', '')
+      if (typeof createdTime === 'string' && createdTime !== '') {
+        const hour = createdTime.slice(0, 2);
+        const minute = createdTime.slice(2, 4);
+        const second = createdTime.slice(4, 6);
+
+        if (hour < 12) {
+          createdTime = `上午${hour}:${minute}:${second}`
+        } else {
+          createdTime = `下午${hour}:${minute}:${second}`
+        }
+      }
+
+      return (
+        <div key={this.props.path + '_details'} className="image-details">
+          <div className="image-values">
+            <FormInput noedit multiline>
+              <h6>檔案位址</h6>
+              <p>{this.getLocation()}</p>
```

Comment:
In this design, how could the user(i.e. photographer) deal with the URL.

Reply:
They don't have to deal with the URL.
Do you think we could omit this field?
I think it's good to be there.

Reply:
We can keep the design and highlight this item during the demo to collect the feedback of the actual users.